can-define.js

```diff
@@ -727,10 +735,19 @@ getDefinitionsAndMethods = function(defines, baseDefines) {
 			return;
 		} else {
 			var result = getDefinitionOrMethod(prop, value, defaultDefinition);
-			if(result && typeof result === "object") {
+			if(result && typeof result === "object" && !isEmptyObject(result)) {
 				definitions[prop] = result;
-			} else {
-				methods[prop] = result;
+			} 
+			else {
+				// Removed adding raw values that are not functions
+				if (typeof result === 'function') {
+					methods[prop] = result;
+				}
+				//!steal-remove-start
+				else if (typeof result !== 'undefined') {
+					canLogDev.error("A Constructor or string must be used with shorthand definitions: https://canjs.com/doc/can-define.types.propDefinition.html#String");
```

Comment:
Can you add the `prop` and `constructor.shortName` to this warning message so that people can track down where the problem is?

Also, it might be better to just say that the given definition does not match any of the supported PropDefinitions and link to the docs.